src/components/common/Heatmap.tsx

```diff
@@ -27,67 +28,76 @@ const getYLabelsAndIds = (list: HierarchicalItem[], prefix = ''): { labels: stri
-                                               onCellClick,
-                                               connectionsMap
-                                           }) => {
-
-    const [yAxis, setYAxis] = useState<HierarchicalItem[]>(initialYAxis);
-    const data = useMemo(() => {
-        return getHeatmapData(yAxis, connectionsMap);
-    }, [yAxis, connectionsMap]);
-    const [selectedCell, setSelectedCell] = useState<{ x: number, y: number } | null>(null);
-
-
-    const handleItemClick = (item: HierarchicalItem) => {
+    xAxis, yAxis, setYAxis,
+    xAxisLabel, yAxisLabel,
+    onCellClick, selectedCell, heatmapData, secondaryHeatmapData
+}) => {
+    const secondary = secondaryHeatmapData ? true : false
+    const heatmapMatrixData = secondary ? prepareSecondaryHeatmapData(secondaryHeatmapData) : heatmapData;
+    const handleCollapseClick = (item: HierarchicalItem) => {
         const updateList = (list: HierarchicalItem[], selectedItem: HierarchicalItem): HierarchicalItem[] => {
-            return list.map(listItem => {
+            return list?.map(listItem => {
                 if (listItem.label === selectedItem.label) {
-                    return {...listItem, expanded: !listItem.expanded};
+                    return { ...listItem, expanded: !listItem.expanded };
                 } else if (listItem.children) {
-                    return {...listItem, children: updateList(listItem.children, selectedItem)};
+                    return { ...listItem, children: updateList(listItem.children, selectedItem) };
                 }
                 return listItem;
             });
         };
-
         const updatedList = updateList(yAxis, item);
         setYAxis(updatedList);
     };
 
-    const handleClick = (x: number, y: number, yIds: string[]): void => {
-        setSelectedCell({x, y});
-        if(onCellClick){
-            onCellClick(x, yIds[y])
+    const handleCellClick = (x: number, y: number) => {
+        const ids = yAxisData.ids
+        if (onCellClick) {
+            onCellClick(x, y, ids[y]);
+        }
+    }
+
+
+    const getCellBgColorFromPhenotype = (
+        normalizedValue: number,
+        _x: number,
+        _y: number
+    ) => {
+        if (secondary && secondaryHeatmapData && secondaryHeatmapData[_y] && secondaryHeatmapData[_y][_x]) {
+            const phenotypeColors = secondaryHeatmapData[_y][_x]?.color;
+
+            const phenotypeColorsWithPercentage = phenotypeColors.map((color, index) => {
+                return `${color} ${100 / phenotypeColors.length * index}%, ${color} ${100 / phenotypeColors.length * (index + 1)}%`
+            })
+            let phenotypeColor = phenotypeColors.length > 1 ? `linear-gradient(to right, ${phenotypeColorsWithPercentage.join(',')}` :
+                phenotypeColors.length === 1 ? phenotypeColors[0] : '';
+            phenotypeColor = phenotypeColor?.replace(/rgba\(([^,]+),([^,]+),([^,]+),([^)]+)\)/g, `rgba($1,$2,$3,${normalizedValue})`).replace(
```

Comment:
suggestion (non-blocking) this should probably be a function since it isn't very readable like this